source/SpinalHDL/Data types/Int.rst

```diff
@@ -416,3 +440,143 @@ Misc
+ Scala.math            round   ROUNDTOUP           round(1.5)=2,round(2.5)=3;round(-1.5)=-1,round(-2.5)=-2   always to +Infinity
+ SpinalHDL             round   ROUNDTOINF          round(1.5)=2,round(2.5)=3;round(-1.5)=-2,round(-2.5)=-3   round to ±Infinity
+===================== ======= =================== ========================================================= =====================
+
+In spinal we chose type of `ROUNDTOINF` as Default RoundType, Api `round = roundToInf`
+
+.. code-block:: scala
+
+   val A  = SInt(16 bit)
+   val B  = A.roundToInf(6 bits) //default 'align = false' with carry, got 11 bit
+   val B  = A.roundToInf(6 bits, align = true) // sat 1 carry bit, got 10 bit
+   val B  = A.floor(6 bits)             //return 10 bit
+   val B  = A.floorToZero(6 bits)       //return 10 bit
+   val B  = A.ceil(6 bits)              //ceil with carry so return 11 bit
+   val B  = A.ceil(6 bits, align =true) //ceil with carry then sat 1 bit return 10 bit
+   val B  = A.ceilToInf(6 bits)
+   val B  = A.roundUp(6 bits)
+   val B  = A.roundDown(6 bits)
+   val B  = A.roundToInf(6 bits)
+   val B  = A.roundToZero(6 bits)
+   val B  = A.round(6 bits)             //spinalHDL use roundToInf as default round
+
+   val B0 = roundToInf(6 bits, align=true)          ---+
+                                                       |--> equal
+   val B1 = roundToInf(6 bits, align=false).sat(1)  ---+
+
+only `floor` and `floorToZero` without align option, they do not need carry bit.
+other round operation need carry bit.
+
+**round Api**
+
+============= =========== ============================ ===================== ====================
+ API           UInt/SInt   description                  Return(align=false)   Return(align=true)
+============= =========== ============================ ===================== ====================
+ floor         Both                                     w(x)-n   bits         w(x)-n bits
+ floorToZero   SInt        equal to floor in UInt       w(x)-n   bits         w(x)-n bits
+ ceil          Both                                     w(x)-n+1 bits         w(x)-n bits
+ ceilToInf     SInt        equal to ceil in UInt        w(x)-n+1 bits         w(x)-n bits
+ roundUp       Both        simple for HW                w(x)-n+1 bits         w(x)-n bits
+ roundDown     Both                                     w(x)-n+1 bits         w(x)-n bits
+ roundToInf    SInt        most Common                  w(x)-n+1 bits         w(x)-n bits
+ roundToZero   SInt        equal to roundDown in UInt   w(x)-n+1 bits         w(x)-n bits
+ round         Both        SpinalHDL chose roundToInf   w(x)-n+1 bits         w(x)-n bits
+============= =========== ============================ ===================== ====================
+
+although `roundToInf` is very common.
+but `roundUp` with lowerest cost and good timing, almost no performance lost.   
+so `roundUp` is very recommended in your work.
```

Comment:
Ok this tre sentence are disconnected from each other, but connected by a `but` and `so`. maybe it should be:
```
.. note::
   | Although `roundToInf` is very common rounding strategy, `roundUp` has the lower logic cost and almost no performance hit.
   | So using `roundUp` is recommended.
```